File: leetcode/editor/en/[9]Palindrome Number.ts

//leetcode submit region begin(Prohibit modification and deletion)
function isPalindrome(x: number): boolean {
    if (x < 0) {
        return false
    }

    if (x < 10) {
        return true
    }

    let reversed = 0
    let normal = x

    while (normal > 0) {
        reversed = (reversed * 10) + (normal % 10)
        normal = Math.floor(normal / 10)
    }

    return x === reversed
};
//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/en/[9]Palindrome Number.py

# leetcode submit region begin(Prohibit modification and deletion)
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False
        if x < 10:
            return True

        return str(x) == str(x)[::-1]
# leetcode submit region end(Prohibit modification and deletion)
